### PR TITLE
Add on-cluster workflow test to CI

### LINKED ISSUE #652: Integrate Argo Workflows into Hera's CI
**Is your feature request related to a problem? Please describe.**
Right now there's no standard way to test Hera workflow examples and users *need* their own clusters for testing. 

**Describe the solution you'd like**
Integrate minikube/k3s into CI and add a job that runs at least several workflows in CI to ensure submission works

**Describe alternatives you've considered**
Not sure there are any that are specifically simpler than integrating k8s into CI but definitely open to any!

**Additional context**
None

### PR DESCRIPTION
**Pull request by elliotgunton**
**Pull Request Checklist**
- [x] Fixes #652
- [x] Tests added
- [x] Documentation/examples added
- [x] [Good commit messages](https://cbea.ms/git-commit/) and/or PR title

**Description of PR**
Introduce local Argo cluster workflow tests to CI, and `make` targets to set up a local cluster for testing (this worked very well on GitHub codespaces, where I could even get a web UI to the cluster).

Added a single test to confirm the workflow runs via `create` and reaches the `succeeded` `status.phase`. We can think about creating a more flexible/extensible framework for adding `on_cluster` tests -
* I'd like to use a subset of our examples
* But we want to assert on more than the `status.phase`, e.g. output values, running order, skipped nodes etc, which would require bespoke tests for each
* It would also be good to have a wrapper for fetching these values, e.g. getting a node requires iterating through the `status.nodes` dictionary, which is keyed by hashes, so you have to check `displayName` from the values.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -126,3 +126,43 @@ regenerate-test-data: install-3.8
 	find examples -name "*.yaml" -type f -delete
 	HERA_REGENERATE=1 make test examples
 	@poetry run python -m pytest -k test_for_missing_examples --runxfail
+
+.PHONY: install-minikube
```
Comment (samj1912) [inline Makefile:130]: We should use k3d instead. It is much quicker to spin up.

File: Makefile [hunk 2]
```diff
@@ -44,7 +44,7 @@ lint:  ## Run a `lint` process on Hera and report problems
 
 .PHONY: test
 test:  ## Run tests for Hera
-	@poetry run python -m pytest --cov-report=term-missing
+	@poetry run python -m pytest --cov-report=term-missing -m "not workflow"
```
Comment (flaviuvadan) [inline Makefile:47]: picky comment - would this flag be more intuitive if it were called "remote" or "remote_in_ci"? A bunch of tests test workflows

Comment (elliotgunton) [reply Makefile:47]: Yeah lol I didn't want to get hung up on the naming, STC! 😆 Will change

File: Makefile (hunk 1 above)
Comment (elliotgunton) [reply Makefile]: Just for posterity (and the curious) - minikube took 41s [here](https://github.com/argoproj-labs/hera/actions/runs/7144548456/job/19458392050?pr=887), and k3d took 49s [here](https://github.com/argoproj-labs/hera/actions/runs/7170762330/job/19524240213?pr=887)

Comment (elliotgunton) [reply Makefile]: Benchmarks from https://www.blueshoe.io/blog/minikube-vs-k3d-vs-kind-vs-getdeck-beiboot/ suggest k3d is faster and more efficient.

**Review by flaviuvadan — APPROVED** [review]
This LGTM! The next iteration can address some of the other concerns you mentioned in the description. Thx again for adding this <3

**Pull request merged**